### ISSUE TITLE
Mistral Tool Call Error: Tool call id has to be defined in serving mode.

### ISSUE THREAD

**Issue by hopkins385** [vercel/ai#2477]
### Description

I get the following error after tool call has finished with mistral provider. 
Other providers such as OpenAi or Anthropic are working fine.

```
ERROR: Tool call id has to be defined in serving mode.

  at <anonymous> (node_modules/@ai-sdk/provider-utils/src/response-handler.ts:58:16)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
  at postToApi (node_modules/@ai-sdk/provider-utils/src/post-to-api.ts:80:28)
  at MistralChatLanguageModel.doStream (node_modules/@ai-sdk/mistral/src/mistral-chat-language-model.ts:213:50)
  at fn (node_modules/ai/core/generate-text/stream-text.ts:211:23)
  at <anonymous> (node_modules/ai/core/telemetry/record-span.ts:18:22)
  at _retryWithExponentialBackoff (node_modules/ai/core/util/retry-with-exponential-backoff.ts:36:12)
  at fn (node_modules/ai/core/generate-text/stream-text.ts:199:11)
  at <anonymous> (node_modules/ai/core/telemetry/record-span.ts:18:22)
  at generateStream (server/api/chat/conversation.post.ts:334:1)
  at async nextAsync (node:internal/streams/from:182:33)
```

### Code example
```js
        const mistral = createMistral({
          apiKey: this.config.mistral.apiKey,
        });
        const model = mistral(this.model);
```

```js
  const availableTools = {
    website: tool({
      description: 'Get information about a website',
      parameters: z.object({
        url: z
          .string()
          .url()
          .refine((url) => url.startsWith('https://'), {
            message: 'URL must start with https://',
          })
          .describe('The URL of the website to get information about'),
      }),
      execute: async function ({ url }) {
        const response = await visitWebsite(url);
        return {
          response,
        };
      },
    }),
```

```js
...
      } else if (chunk.type === 'finish' && chunk.finishReason === 'tool-calls') {
        const toolCalls = await initialResult.toolCalls;
        const toolResults = await initialResult.toolResults;

        if (toolResults.length === 0) {
          logger.error('No tool results found');
          continue;
        }

        const toolMessages = [
          {
            role: 'assistant',
            content: toolCalls,
          },
          {
            role: 'tool',
            content: toolResults,
          },
        ];

        const followUpResult = await streamText({
          abortSignal: payload.signal,
          model,
          system: payload.systemPrompt,
          messages: [...payload.messages, ...toolMessages],
          maxTokens: payload.maxTokens,
          tools: availableTools,
        });
...
}
```

### Additional context

Node: v20.14.0
macOS: 14.5 (23F79)

**Comment by lgrammel** [comment]
Reproduced. The Mistral API must have changed.

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2481

**Comment by lgrammel** [comment]
Fixed in `@ai-sdk/mistral@0.0.26`

**Comment by hopkins385** [comment]
Confirmed. Thanks for the fix in no time!

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete provider regression in existing tool-calling behavior that caused requests to fail unexpectedly, and it was subsequently fixed.

### Evidence
- Mistral tool calls failed after completion with the runtime error: "Tool call id has to be defined in serving mode."
- A maintainer reproduced the failure and attributed it to a Mistral API change.
- The issue was resolved by PR #2481 and released in @ai-sdk/mistral@0.0.26; the reporter confirmed the fix.
